presto-main/src/main/java/com/facebook/presto/spiller/Spiller.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spiller;
+
+import com.facebook.presto.spi.Page;
+
+import java.io.Closeable;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public interface Spiller
+        extends Closeable
+{
+    /**
+     * Initiate spilling of pages stream. Returns completed future once spilling has finished.
+     */
+    CompletableFuture<?> spill(Iterator<Page> pageIterator);
+
+    /**
+     * Returns list of previously spilled Pages streams.
+     */
+    List<Iterator<Page>> getSpills();
```

Comment:
This shouldn't be that big an issue as for writing, because in writing we have huge chunks of data that we want to write at once. For reading, we are reading Page by Page from different Iterators one by one - one call to `HashAggregationOperator::getOutput()` should request at most for only couple of Page reads from disks. Read ahead should mitigate this issue even further.

What's more is that implementing this in async fashion is not that simple. It would have to be `List<Iterator<Future<Page>>>` and whole `MergeSorter` would have to handle those Futures. Maybe that would force us to write our own `MergeSort`.

I'm not saying this shouldn't be done eventually, but it doesn't seem as the highest priority at the moment. Similarly as Presto was able to live for very long time with synchronous `PageSinks` and `PageSources` (btw, don't that vast majority of connectors, if not even all of the open source ones, still use synchronous `PageSinks/Sources`?).